### PR TITLE
Add doc examples

### LINKED ISSUE #29347: API Docs: cmp
https://github.com/rust-lang/rust/issues/29329

http://doc.rust-lang.org/std/cmp/

Special to me. This is where I got started working on Rust. :heart:

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #29347

r? @steveklabnik

### CODE REVIEW

File: src/libcore/cmp.rs [hunk 1]
```diff
@@ -14,6 +14,21 @@
 //! by the compiler to implement comparison operators. Rust programs may
 //! implement `PartialOrd` to overload the `<`, `<=`, `>`, and `>=` operators,
 //! and may implement `PartialEq` to overload the `==` and `!=` operators.
+//!
+//! # Examples
+//!
+//! ```
+//! let x : u32 = 0;
```
Comment (steveklabnik) [inline src/libcore/cmp.rs]: standard rust style has no space before the `:`, can you modify this in all the examples?

Also, it's a bit strange to annotate the types here, is there a specific reason?

Comment (GuillaumeGomez) [inline src/libcore/cmp.rs]: > standard rust style has no space before the :, can you modify this in all the examples?

Indeed, I'll update it in all my changes.

> Also, it's a bit strange to annotate the types here, is there a specific reason?

I wanted to make the example as easy to understand as possible. Like this, there is no debate on the type and tkey can focus on what I want to show them.

**Comment by GuillaumeGomez** [comment]
Updated.

**Pull request merged**